tests/typecheck/managers/querysets/test_as_manager.yml

```diff
@@ -146,6 +146,32 @@
                 class MyModel(models.Model):
                     objects = ManagerFromModelQuerySet
 
+-   case: handles_subclasses_of_queryset
+    main: |
+        from myapp.models import MyModel
+        reveal_type(MyModel.objects.example())  # N: Revealed type is "myapp.models.MyModel"
+    installed_apps:
+        - myapp
+    files:
+        -   path: myapp/__init__.py
+        -   path: myapp/models.py
+            content: |
+                from typing import TypeVar
+
+                from django.db import models
+
+                _CTE = TypeVar("_CTE", bound=models.Model)
```

Comment:
I also think we should ensure the upper bound of `QuerySet`s `TypeVar` is honored. So that we know it'll emit an error correctly.

e.g. use a `TypeVar` without upper bound and test pass e.g. `int` as argument

Reply:
I have set : 
class OtherQuerySet(_MyModelQuerySet[str, "MyOtherModel"]):

and I have the error :
myapp/models:30: error: Type argument "str" of "_MyModelQuerySet" must be a subtype of "Model" (diff

Should I create a test for this ?

Reply:
I would be fine if you'd include that error in your tests

Reply:
yes, done :)